docs/adr/0016-use-graphql-and-apollo-for-settings-redesign.md

```diff
@@ -0,0 +1,135 @@
+- If we later choose to use GraphQL, even with the option B approach, we'll face a lot of refactoring at a later time that could be mitigated now.
+
+### Options B & C - use GraphQL and Apollo
+
+#### Description
+
+This section outlines the pros and cons for using GraphQL and Apollo in general and applies to both options B and C.
+
+While GraphQL will be novel tech to the FxA stack, it's not entirely novel to the FxA ecosystem. That is to say, the Admin Panel project (`fxa-admin-panel`/`fxa-admin-server` packages) served as a prototype for experimenting with GQL and Apollo.
+
+##### Pros
+
+- We can use GQL for all Settings Redesign needs and won't need to mix GQL and REST.
+- GQL can consolidate network requests. Even if the GQL queries need to go through the auth server, the client will be relieved of a network request burden and instead, a single request from the client lets the server perform all of the requests. This will offer a performant boost especially on the main Settings page, where currently around 17 different requests must be made.
+- Since only required data will be requested, the data returned won't be overfetched like it is in traditional REST architecture, resulting in a smaller payload.
+- GQL offers a flexible and delcarative approach to data - both FE and BE developers can focus on describing the data rather than implementing and optimizing numerous REST endpoints. This makes it painfully clear what's being sent and received between client and server which can improve visibility into what's happening in components.
+- GraphQL and Apollo have gained a lot of popularity over the years and now have substantial company and community support.
+- This stack promotes strong end-to-end typing.
+- Apollo offers great features out of the box like caching, error handling, optimistic rendering, and nice integration tools for React (and even provides React hooks).
+- The React Settings application will only have to manage interacting with a single endpoint and Apollo Client makes managing the data in our React app straightforward. Apollo Client creates its own internal Redux store under the hood and allows us to store local data in the Apollo cache, the same place that data from network requests will be stored.
+- Apollo also provides a "GraphQL playground" which is a graphical, interactive, in-browser GQL IDE. It not only provides easy to access schemas (and is "self-documenting" in a way), but it allows developers to write queries or mutations to send to the server so the developer can verify it behaves as expected. This can then be copied directly over to where it needs to be used in the application.
+- CRUD operations have been simplified to simply queries and mutations. CRUD in GraphQL becomes "MQMM" and developers don't have to think about a C/U/D differentiation outside of writing the mutation for them.
```

Comment:
Might be beneficial to link up or describe MQMM (and to a certain extent, CRUD, but less important).

Reply:
I'm actually not even sure this is a "pro"... Maybe it should be neutral. It's kind of a weird thing to get used to but I like the simplification.